### ISSUE TITLE
Query timed out after PT2S

### ISSUE THREAD

**Issue by grib0** [glowroot/glowroot#1022]
Starting a new version of glowroot 0.14.0, I got the following exception when it creates the database.

```
glowroot     | 2023-06-27 09:09:10.933 INFO  org.glowroot - Glowroot version: 0.14.0, built 2023-05-14 11:36:22 +0000
glowroot     | 2023-06-27 09:09:10.936 INFO  org.glowroot - Java version: 17.0.6
glowroot     | 2023-06-27 09:09:11.252 INFO  c.d.o.d.i.c.DefaultMavenCoordinates - DataStax Java driver for Apache Cassandra(R) (com.datastax.oss:java-driver-core) version 4.15.0
glowroot     | 2023-06-27 09:09:12.607 WARN  c.d.o.d.i.core.session.PoolManager - [s0] Detected a keyspace change at runtime (<none> => glowroot_local). This is an anti-pattern that should be avoided in production (see 'advanced.request.warn-if-set-keyspace' in the configuration).
glowroot     | 2023-06-27 09:09:12.726 INFO  org.glowroot - connected to Cassandra (version 3.11.14), using keyspace 'glowroot_local' (replication factor 1) and consistency level QUORUM
glowroot     | 2023-06-27 09:09:12.774 INFO  org.glowroot - creating glowroot central schema...
glowroot     | 2023-06-27 09:09:15.562 ERROR org.glowroot - Query timed out after PT2S
glowroot     | com.datastax.oss.driver.api.core.DriverTimeoutException: Query timed out after PT2S
glowroot     |  at com.datastax.oss.driver.api.core.DriverTimeoutException.copy(DriverTimeoutException.java:34)
glowroot     |  at com.datastax.oss.driver.internal.core.util.concurrent.CompletableFutures.getUninterruptibly(CompletableFutures.java:149)
glowroot     |  at com.datastax.oss.driver.internal.core.cql.CqlRequestSyncProcessor.process(CqlRequestSyncProcessor.java:53)
glowroot     |  at com.datastax.oss.driver.internal.core.cql.CqlRequestSyncProcessor.process(CqlRequestSyncProcessor.java:30)
glowroot     |  at com.datastax.oss.driver.internal.core.session.DefaultSession.execute(DefaultSession.java:230)
glowroot     |  at com.datastax.oss.driver.api.core.cql.SyncCqlSession.execute(SyncCqlSession.java:54)
glowroot     |  at com.datastax.oss.driver.api.core.cql.SyncCqlSession.execute(SyncCqlSession.java:78)
glowroot     |  at org.glowroot.central.util.Session.updateSchemaWithRetry(Session.java:376)
glowroot     |  at org.glowroot.central.util.Session.updateSchemaWithRetry(Session.java:305)
glowroot     |  at org.glowroot.central.util.Session.createTableWithTracking(Session.java:320)
glowroot     |  at org.glowroot.central.util.Session.createTableWithTWCS(Session.java:257)
glowroot     |  at org.glowroot.central.util.Session.createTableWithTWCS(Session.java:237)
glowroot     |  at org.glowroot.central.util.Session.createTableWithTWCS(Session.java:232)
glowroot     |  at org.glowroot.central.repo.AggregateDaoImpl.<init>(AggregateDaoImpl.java:303)
glowroot     |  at org.glowroot.central.repo.CentralRepoModule.<init>(CentralRepoModule.java:138)
glowroot     |  at org.glowroot.central.CentralModule.<init>(CentralModule.java:192)
glowroot     |  at org.glowroot.central.CentralModule.create(CentralModule.java:130)
glowroot     |  at org.glowroot.central.CentralModule.create(CentralModule.java:125)
glowroot     |  at org.glowroot.central.Main.start(Main.java:44)
glowroot     |  at org.glowroot.central.Main.main(Main.java:28)
glowroot     | Exception in thread "main" com.datastax.oss.driver.api.core.DriverTimeoutException: Query timed out after PT2S
glowroot     |  at com.datastax.oss.driver.api.core.DriverTimeoutException.copy(DriverTimeoutException.java:34)
glowroot     |  at com.datastax.oss.driver.internal.core.util.concurrent.CompletableFutures.getUninterruptibly(CompletableFutures.java:149)
glowroot     |  at com.datastax.oss.driver.internal.core.cql.CqlRequestSyncProcessor.process(CqlRequestSyncProcessor.java:53)
glowroot     |  at com.datastax.oss.driver.internal.core.cql.CqlRequestSyncProcessor.process(CqlRequestSyncProcessor.java:30)
glowroot     |  at com.datastax.oss.driver.internal.core.session.DefaultSession.execute(DefaultSession.java:230)
glowroot     |  at com.datastax.oss.driver.api.core.cql.SyncCqlSession.execute(SyncCqlSession.java:54)
glowroot     |  at com.datastax.oss.driver.api.core.cql.SyncCqlSession.execute(SyncCqlSession.java:78)
glowroot     |  at org.glowroot.central.util.Session.updateSchemaWithRetry(Session.java:376)
glowroot     |  at org.glowroot.central.util.Session.updateSchemaWithRetry(Session.java:305)
glowroot     |  at org.glowroot.central.util.Session.createTableWithTracking(Session.java:320)
glowroot     |  at org.glowroot.central.util.Session.createTableWithTWCS(Session.java:257)
glowroot     |  at org.glowroot.central.util.Session.createTableWithTWCS(Session.java:237)
glowroot     |  at org.glowroot.central.util.Session.createTableWithTWCS(Session.java:232)
glowroot     |  at org.glowroot.central.repo.AggregateDaoImpl.<init>(AggregateDaoImpl.java:303)
glowroot     |  at org.glowroot.central.repo.CentralRepoModule.<init>(CentralRepoModule.java:138)
glowroot     |  at org.glowroot.central.CentralModule.<init>(CentralModule.java:192)
glowroot     |  at org.glowroot.central.CentralModule.create(CentralModule.java:130)
glowroot     |  at org.glowroot.central.CentralModule.create(CentralModule.java:125)
glowroot     |  at org.glowroot.central.Main.start(Main.java:44)
glowroot     |  at org.glowroot.central.Main.main(Main.java:28)

```

I need to restart multiple times in order to get the database set up.
After this multiple restarts, I did not get error and glowroot works fine.

Cassandra version : 3.11.14

As the cassandra java driver reference document mention, we should be able to add the system property but it doesn't work
https://docs.datastax.com/en/developer/java-driver/4.16/manual/core/configuration/

`GLOWROOT_OPTS: -Ddatastax-java-driver.basic.request.timeout="5 seconds"`

**Comment by MikaMLehtonen** [comment]
Same problem, but in normal driving conditions, causes lot of errors in the log and in the UI.

**Comment by MikaMLehtonen** [comment]
0.14.1-beta.2 didn't fix this issue in my case.

**Comment by MikaMLehtonen** [comment]
2023-07-25 09:29:31.234 ERROR o.g.central.CollectorServiceImpl - xxxxxxx::Nodes::xxx.yyy.zzz.www-node-1 - Query timed out after PT2S
com.datastax.oss.driver.api.core.DriverTimeoutException: Query timed out after PT2S
        at com.datastax.oss.driver.api.core.DriverTimeoutException.copy(DriverTimeoutException.java:34)
        at org.glowroot.central.util.MoreFutures.unwrapDriverException(MoreFutures.java:82)
        at org.glowroot.central.util.MoreFutures.waitForAll(MoreFutures.java:39)
        at org.glowroot.central.repo.GaugeValueDaoImpl.store(GaugeValueDaoImpl.java:214)
        at org.glowroot.central.repo.GaugeValueDaoImpl.store(GaugeValueDaoImpl.java:183)
        at org.glowroot.central.CollectorServiceImpl.collectGaugeValuesUnderThrottle(CollectorServiceImpl.java:384)
        at org.glowroot.central.CollectorServiceImpl$3.run(CollectorServiceImpl.java:281)
        at org.glowroot.central.CollectorServiceImpl.throttle(CollectorServiceImpl.java:316)
        at org.glowroot.central.CollectorServiceImpl.throttledCollectGaugeValues(CollectorServiceImpl.java:277)
        at org.glowroot.central.CollectorServiceImpl.collectGaugeValues(CollectorServiceImpl.java:209)
        at org.glowroot.wire.api.model.CollectorServiceGrpc$MethodHandlers.invoke(CollectorServiceGrpc.java:584)
        at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:182)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:346)
        at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:860)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)

**Comment by john9x** [comment]
>I have the same issue with 0.14.1-beta.2

+1
```
2023-08-25 06:05:09.281 INFO  org.glowroot - Glowroot version: 0.14.1-beta.2, built 2023-07-22 21:55:30 +0000
2023-08-25 06:05:09.289 INFO  org.glowroot - Java version: 17.0.7
2023-08-25 06:05:09.938 INFO  c.d.o.d.i.c.DefaultMavenCoordinates - DataStax Java driver for Apache Cassandra(R) (com.datastax.oss:java-driver-core) version 4.16.0
2023-08-25 06:05:10.196 INFO  c.d.o.d.i.c.c.CqlPrepareAsyncProcessor - Adding handler to invalidate cached prepared statements on type changes
2023-08-25 06:05:11.525 WARN  c.d.o.d.i.core.session.PoolManager - [s0] Detected a keyspace change at runtime (<none> => glowroot). This is an anti-pattern that should be avoided in production (see 'advanced.request.warn-if-set-keyspace' in the configuration).
2023-08-25 06:05:11.857 INFO  org.glowroot - connected to Cassandra (version 3.11.15), using keyspace 'glowroot' (replication factor 1) and consistency level QUORUM

...

2023-08-25 06:05:33.524 ERROR o.g.central.CollectorServiceImpl -  - com.datastax.oss.driver.api.core.DriverTimeoutException: Query timed out after PT2S
java.util.concurrent.CompletionException: com.datastax.oss.driver.api.core.DriverTimeoutException: Query timed out after PT2S
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:874)
        at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
        at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.datastax.oss.driver.api.core.DriverTimeoutException: Query timed out after PT2S
        at com.datastax.oss.driver.internal.core.cql.CqlRequestHandler.lambda$scheduleTimeout$1(CqlRequestHandler.java:207)
        at io.netty.util.HashedWheelTimer$HashedWheelTimeout.run(HashedWheelTimer.java:715)
        at io.netty.util.concurrent.ImmediateExecutor.execute(ImmediateExecutor.java:34)
        at io.netty.util.HashedWheelTimer$HashedWheelTimeout.expire(HashedWheelTimer.java:703)
        at io.netty.util.HashedWheelTimer$HashedWheelBucket.expireTimeouts(HashedWheelTimer.java:790)
        at io.netty.util.HashedWheelTimer$Worker.run(HashedWheelTimer.java:503)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        ... 1 common frames omitted

```

**Comment by john9x** [comment]
Also I see a lot of
```
2023-08-25 06:08:17.874 ERROR o.g.central.CollectorServiceImpl - api-a1.us.navixy - com.datastax.oss.driver.api.core.AllNodesFailedException: All 1 node(s) tried for the query failed (showing first 1 nodes, use getAllErrors() for more): Node(endPoint=cassandra/172.18.0.2:9042, hostId=65aad6e1-7a26-4eb0-80b8-d84c5e257f19, hashCode=389271b0): [com.datastax.oss.driver.api.core.NodeUnavailableException: No connection was available to Node(endPoint=cassandra/172.18.0.2:9042, hostId=65aad6e1-7a26-4eb0-80b8-d84c5e257f19, hashCode=389271b0)]
java.util.concurrent.CompletionException: com.datastax.oss.driver.api.core.AllNodesFailedException: All 1 node(s) tried for the query failed (showing first 1 nodes, use getAllErrors() for more): Node(endPoint=cassandra/172.18.0.2:9042, hostId=65aad6e1-7a26-4eb0-80b8-d84c5e257f19, hashCode=389271b0): [com.datastax.oss.driver.api.core.NodeUnavailableException: No connection was available to Node(endPoint=cassandra/172.18.0.2:9042, hostId=65aad6e1-7a26-4eb0-80b8-d84c5e257f19, hashCode=389271b0)]
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:874)
        at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
        at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.datastax.oss.driver.api.core.AllNodesFailedException: All 1 node(s) tried for the query failed (showing first 1 nodes, use getAllErrors() for more): Node(endPoint=cassandra/172.18.0.2:9042, hostId=65aad6e1-7a26-4eb0-80b8-d84c5e257f19, hashCode=389271b0): [com.datastax.oss.driver.api.core.NodeUnavailableException: No connection was available to Node(endPoint=cassandra/172.18.0.2:9042, hostId=65aad6e1-7a26-4eb0-80b8-d84c5e257f19, hashCode=389271b0)]
        at com.datastax.oss.driver.api.core.AllNodesFailedException.fromErrors(AllNodesFailedException.java:55)
        at com.datastax.oss.driver.internal.core.cql.CqlRequestHandler.sendRequest(CqlRequestHandler.java:264)
        at com.datastax.oss.driver.internal.core.cql.CqlRequestHandler.onThrottleReady(CqlRequestHandler.java:195)
        at com.datastax.oss.driver.internal.core.session.throttling.ConcurrencyLimitingRequestThrottler.register(ConcurrencyLimitingRequestThrottler.java:97)
        at com.datastax.oss.driver.internal.core.cql.CqlRequestHandler.<init>(CqlRequestHandler.java:172)
        at com.datastax.oss.driver.internal.core.cql.CqlRequestAsyncProcessor.process(CqlRequestAsyncProcessor.java:44)
        at com.datastax.oss.driver.internal.core.cql.CqlRequestAsyncProcessor.process(CqlRequestAsyncProcessor.java:29)
        at com.datastax.oss.driver.internal.core.session.DefaultSession.execute(DefaultSession.java:230)
        at com.datastax.oss.driver.api.core.cql.AsyncCqlSession.executeAsync(AsyncCqlSession.java:42)
        at org.glowroot.central.util.Session.lambda$writeAsync$3(Session.java:186)
        at org.glowroot.central.util.Session.throttle(Session.java:356)
        at org.glowroot.central.util.Session.throttleWrite(Session.java:345)
        at org.glowroot.central.util.Session.writeAsync(Session.java:182)
        at org.glowroot.central.repo.FullQueryTextDao.store(FullQueryTextDao.java:114)
        at org.glowroot.central.repo.TraceDaoImpl.storeInternal(TraceDaoImpl.java:625)
        at org.glowroot.central.repo.TraceDaoImpl.store(TraceDaoImpl.java:594)
        at org.glowroot.central.repo.TraceDaoImpl.store(TraceDaoImpl.java:584)
        at org.glowroot.central.CollectorServiceImpl.collectTraceUnderThrottle(CollectorServiceImpl.java:442)
        at org.glowroot.central.CollectorServiceImpl$4.run(CollectorServiceImpl.java:291)
        at org.glowroot.central.CollectorServiceImpl.throttle(CollectorServiceImpl.java:316)
        at org.glowroot.central.CollectorServiceImpl.throttledCollectTrace(CollectorServiceImpl.java:288)
        at org.glowroot.central.CollectorServiceImpl$TraceStreamObserver.onCompletedInternal(CollectorServiceImpl.java:764)
        at org.glowroot.central.CollectorServiceImpl$TraceStreamObserver.onCompleted(CollectorServiceImpl.java:686)
        at io.grpc.stub.ServerCalls$StreamingServerCallHandler$StreamingServerCallListener.onHalfClose(ServerCalls.java:273)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:346)
        at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:860)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        ... 1 common frames omitted
        Suppressed: com.datastax.oss.driver.api.core.NodeUnavailableException: No connection was available to Node(endPoint=cassandra/172.18.0.2:9042, hostId=65aad6e1-7a26-4eb0-80b8-d84c5e257f19, hashCode=389271b0)
                at com.datastax.oss.driver.internal.core.cql.CqlRequestHandler.sendRequest(CqlRequestHandler.java:256)
                ... 29 common frames omitted
```

I think there is some problem with Cassandra performance but I'm not familiar with Cassandra tuning.

**Comment by Nowheresly** [comment]
I would recommend to switch to the latest version of glowroot available [here](https://github.com/glowroot/glowroot/releases/tag/v0.14.1-beta.6).
Your issue should now be fixed.
